server/src/controllers/user.controller.ts

```diff
@@ -21,43 +21,49 @@ class UserController {
     this.router.get(`${this.prefix}/:id`, (req: Request, res: Response) =>
       this.getUserById(req, res)
     );
-    this.router.post(this.prefix, (req: Request, res: Response) =>
-      this.createUsers(req, res)
+    this.router.put(`${this.prefix}/:id`, (req: Request, res: Response) =>
+      this.updateUserById(req, res)
+    );
+    this.router.delete(`${this.prefix}/:id`, (req: Request, res: Response) =>
+      this.deleteUserById(req, res)
     );
   }
 
-  private getUsers(req: Request, res: Response) {
-    let users = this.userService.getUsers();
+  private async getUsers(req: Request, res: Response) {
+    let users = await this.userService.getUsers();
 
     return new SuccessResult({
       msg: Result.transformRequestOnMsg(req),
       data: users,
     }).handleSuccess(res);
   }
 
-  private getUserById(req: Request, res: Response) {
-    let user = this.userService.getUserById(req.params.id);
+  private async getUserById(req: Request, res: Response) {
+    let user = await this.userService.getUserById(req.params.id);
 
     return new SuccessResult({
       msg: Result.transformRequestOnMsg(req),
       data: user,
```

Comment:
When you send the user as the response data, do you also send the password? If that's the case, it could become a security issue. It's recommended to only send the user's information, except for the password.